concrete/src/Http/Middleware/ThumbnailMiddleware.php

```diff
@@ -60,20 +63,23 @@ public function process(Request $request, DelegateInterface $frame)
         $response = $frame->next($request);
 
         if ($this->app->isInstalled()) {
-            if ($response->getStatusCode() == 200) {
-                /* @var Connection $database */
-                try {
-                    $database = $this->getConnection();
-                } catch (\InvalidArgumentException $e) {
-                    // Don't die here if there's no available database connection
-                    $database = null;
-                }
-
-                if ($database) {
+            $responseStatusCode = (int) $response->getStatusCode();
+            if ($responseStatusCode === 200 || $responseStatusCode === 404) {
```

Comment:
I wonder if we shouldn't have a different flag for these. Perhaps we should set a header in the response that makes it clear that thumbnails can be generated. That way we can just generate thumbnails in the dashboard or when you're logged in